scrapy/utils/url.py

```diff
@@ -114,10 +115,34 @@ def escape_ajax(url):
 
 def add_http_if_no_scheme(url):
     """Add http as the default scheme if it is missing from the url."""
-    if url.startswith('//'):
-        url = 'http:' + url
-        return url
-    parser = parse_url(url)
-    if not parser.scheme or not parser.netloc:
-        url = 'http://' + url
+    match = re.match(r"^\w+://", url, flags=re.I)
+    if not match:
+        parts = urlparse(url)
+        scheme = "http:" if parts.netloc else "http://"
+        url = scheme + url
+
     return url
+
+
+def guess_scheme(url):
+    """Given an URL as string,
+    returns a FileURI if it looks like a file path,
+    otherwise returns an HTTP URL
```

Comment:
What about `Add an URL scheme if missing: file:// for filepath-like input or http:// otherwise.`?

`Add an URL scheme if missing, returning either a File URI for filepath-like input or HTTP URL otherwise` sounds a bit ambiguous - it is not clear if file:// or http:// url is returned always, or only if scheme is missing.